.github/workflows/ci.yml

```diff
@@ -12,28 +12,17 @@ jobs:
     strategy:
       matrix:
         os: [macos-latest, ubuntu-latest, windows-latest]
-        node_version: [10, 12]
+        node_version: [14, 16]
     name: ${{ matrix.os }} / Node v${{ matrix.node_version }}
     runs-on: ${{ matrix.os }}
     steps:
       - run: git config --global core.autocrlf false
       - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node_version }}
-      - name: Get yarn cache directory path
-        id: yarn-cache-dir-path
-        run: echo "::set-output name=dir::$(yarn cache dir)"
-      - name: Restore yarn cache
-        uses: actions/cache@v1
-        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
-        with:
-          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
```

Comment:
To take advantage of yarn's cache, you can:
* Commit the `.yarn/cache` folder files, and
* Add the `--immutable-cache` to the `install`, that will use the local cache files: `yarn install --immutable --immutable-cache`. If there are no cached files, it will fail. This is an easy way to have a replicable state in the repository without hitting the network.

I've ommited this here, because I think that may be opinionated, and would flood the PR with a lot of zip files.

Reply:
Does this remove our cache completely as the change is currently?

Reply:
Yes, but probably I underestimated how valuable that feature is.
Investigatint the latest status of the cache action, I've found this: https://github.com/actions/setup-node#caching-packages-dependencies
If I understand it ok, seems like actions/setup-node@v2 supports caching `yarn` dependencies (along with `npm` and others).
I'll use it here!

Reply:
Seems to work:
* I've forced the actions to run twice.
* The first time (https://github.com/Fitbit/fitbit-sdk-toolchain/actions/runs/1701942365), every one of them downloaded the dependencies (installation took ~1 min)
* The second time (https://github.com/Fitbit/fitbit-sdk-toolchain/runs/4827540336), they found the cache and restored it (installation took ~7 sec, as previously)

```
Run actions/setup-node@v2
  with:
    node-version: 16
    cache: yarn
    always-auth: false
    check-latest: false
    token: ***
Found in cache @ /opt/hostedtoolcache/node/16.13.2/x64
/usr/local/bin/yarn --version
3.1.1
/usr/local/bin/yarn config get cacheFolder
/home/runner/work/fitbit-sdk-toolchain/fitbit-sdk-toolchain/.yarn/cache
Received 41411228 of 41411228 (100.0%), 43.4 MBs/sec
Cache Size: ~39 MB (41411228 B)
/usr/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/3ea0f9fb-4b1d-4b74-9b77-22639a562845/cache.tzst -P -C /home/runner/work/fitbit-sdk-toolchain/fitbit-sdk-toolchain
Cache restored successfully

Cache restored from key: node-cache-Linux-yarn-82d81f2eea8323141a948e36f6f276f863b91345d6c57c66fdf04d93242a60b0
```